src/components/common/dashboard/Dashboard.vue

```diff
@@ -0,0 +1,324 @@
+                        transactionName: UC4Identifier.TRANSACTION_ADD_MATRICULATION,
+                        parameters: "SS2020 Computer Science v3",
+                    },
+                    state: OperationStatus.REJECTED,
+                    reason: "Mocked Reason",
+                    existingApprovals: {
+                        users: [],
+                        groups: [],
+                    } as ApprovalList,
+                    missingApprovals: {
+                        users: [],
+                        groups: ["Admin"],
+                    } as ApprovalList,
+                } as Operation,
+                {
+                    operationId: "FinishedMatriculation1",
+                    initiator: "MockUser4",
+                    initiatedTimestamp: "2011-10-05T14:48:00.000Z",
+                    lastModifiedTimestamp: "2020-10-05T16:48:00.000Z",
+                    transactionInfo: {
+                        contractName: UC4Identifier.CONTRACT_MATRICULATION,
+                        transactionName: UC4Identifier.TRANSACTION_ADD_MATRICULATION,
+                        parameters: "SS2020 Computer Science v3",
+                    },
+                    state: OperationStatus.FINISHED,
+                    reason: "",
+                    existingApprovals: {
+                        users: [],
+                        groups: ["Admin"],
+                    } as ApprovalList,
+                    missingApprovals: {
+                        users: [],
+                        groups: [],
+                    } as ApprovalList,
+                } as Operation,
+            ];
+
+            let mockedWatchedOps = [
+                {
+                    operationId: "FinishedMatriculation1",
+                    initiator: "MockUser4",
+                    initiatedTimestamp: "2011-10-05T14:48:00.000Z",
+                    lastModifiedTimestamp: "2020-10-05T16:48:00.000Z",
+                    transactionInfo: {
+                        contractName: UC4Identifier.CONTRACT_MATRICULATION,
+                        transactionName: UC4Identifier.TRANSACTION_ADD_MATRICULATION,
+                        parameters: "SS2020 Computer Science v3",
+                    },
+                    state: OperationStatus.FINISHED,
+                    reason: "",
+                    existingApprovals: {
+                        users: [],
+                        groups: ["Admin"],
+                    } as ApprovalList,
+                    missingApprovals: {
+                        users: [],
+                        groups: [],
+                    } as ApprovalList,
+                } as Operation,
+            ];
+
+            onBeforeMount(async () => {
+                await refresh();
+                await getEnrollmentId();
+                await getRole();
+            });
+
+            const isAdmin = computed(() => {
+                return role.value == Role.ADMIN;
```

Comment:
instead of role, we can use the store here

Reply:
We need the role twice in this component. Hence using an explicit getter function and a variable for it seems to be fine